### PR TITLE
profiles: replace minimal with local

### PR DESCRIPTION
**Pull request by pbrezina**
This pull request implements changes requested and approved in:
https://fedoraproject.org/wiki/Changes/SSSDRemoveFilesProvider

- New “local” profile to handle local users without SSSD will be introduced. This profile will be based on “minimal”, but it may gain more features.
- “minimal” profile will be removed and replaced by “local”. If system has minimal profile selected, it will automatically switch to local.
- “Local” profile will be now the default profile
- ”sssd” profile will loose with-files-domain and with-files-access-provider options, and will gain --with-tlog option

### CODE REVIEW

**Comment by pbrezina** [comment]
@alexey-tikhonov @justin-stephenson

File: profiles/sssd/README [hunk 1]
```diff
@@ -106,6 +106,10 @@ with-libvirt::
     Enable connecting to libvirt VMs using the hostname configured in the
     guest OS or, as a fallback, their name.
 
+with-tlog::
+    Enable support for automatic tlog session recordings in cooperation with
```
Comment (alexey-tikhonov) [inline profiles/sssd/README:110]: I would remove "automatic", it's misleading, imo.

File: profiles/local/README [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+without-nullok::
+    Do not add nullok parameter to pam_unix.
+
+EXAMPLES
+--------
+
+* Enable minimal profile
+
+  authselect select minimal
```
Comment (alexey-tikhonov) [inline profiles/local/README]: `minimal` -> `local`

File: profiles/local/smartcard-auth [hunk 3]
```diff
@@ -0,0 +1 @@
+auth required pam_debug.so auth=authinfo_unavail
```
Comment (alexey-tikhonov) [inline profiles/local/smartcard-auth]: Why is this file required?

File: rpm/authselect.spec.in [hunk 4]
```diff
@@ -321,6 +321,16 @@ if [ -f %{forcefile} ]; then
     %__rm -f %{forcefile}
 fi
 
+# Minimal profile was removed. Switch to local.
+%__sed -i '1 s/^minimal$/local/'  %{_sysconfdir}/authselect/authselect.conf
```
Comment (alexey-tikhonov) [inline rpm/authselect.spec.in:325]: Add a suffix to create a backup?

File: profiles/sssd/README (hunk 1 above)
Comment (pbrezina) [reply profiles/sssd/README:110]: Done.

File: profiles/local/README (hunk 2 above)
Comment (pbrezina) [reply profiles/local/README]: Done.

File: profiles/local/smartcard-auth (hunk 3 above)
Comment (pbrezina) [reply profiles/local/smartcard-auth]: If someone enables smartcard auth in gdm using dconf, they will get a proper message.

Comment (pbrezina) [reply profiles/local/smartcard-auth]: And any authentication attempt will fail of course.

File: rpm/authselect.spec.in (hunk 4 above)
Comment (pbrezina) [reply rpm/authselect.spec.in:325]: Nah, I don't think it it needed here. It would just create a file that would be on the system forever. If by any chance it gets corrupted, users will run "authselect select --force" anyway instead of restoring from a backup file that hey are not aware of (and that has outdated content so they will have to run select anyway).

**Review by alexey-tikhonov — APPROVED** [review]

**Comment by justin-stephenson** [comment]
Hi @pbrezina do I miss something ?

~~~
justin@fedora:~$ sudo dnf copr enable @authselect/pr359 
Enabling a Copr repository. Please note that this repository is not part
of the main distribution, and quality may vary.

The Fedora Project does not exercise any power over the contents of
this repository beyond the rules outlined in the Copr FAQ at
<https://docs.pagure.org/copr.copr/user_documentation.html#what-i-can-build-in-copr>,
and packages are not held to any quality or security level.

Please do not file bug reports about these packages in Fedora
Bugzilla. In case of problems, contact the owner of this repository.

Do you really want to enable copr.fedorainfracloud.org/@authselect/pr359? [y/N]: y
Repository successfully enabled.
justin@fedora:~$ cat /etc/redhat-release 
Fedora release 39 (Thirty Nine)
justin@fedora:~$ rpm -q authselect
authselect-1.4.3-1.fc39.x86_64
justin@fedora:~$ sudo dnf update authselect
Copr repo for pr359 owned by @authselect                                                                                                                                                                                                         18 kB/s | 1.5 kB     00:00    
Dependencies resolved.
Nothing to do.
Complete!
justin@fedora:~$ sudo dnf downgrade authselect
Last metadata expiration check: 0:00:05 ago on Mon 15 Jan 2024 11:29:46 AM EST.
Error: 
 Problem: problem with installed package pam-1.5.3-3.fc39.x86_64
  - package pam-1.5.3-3.fc39.x86_64 from @System requires authselect >= 1.3, but none of the providers can be installed
  - package pam-1.5.3-2.fc39.x86_64 from fedora requires authselect >= 1.3, but none of the providers can be installed
  - package pam-1.5.3-3.fc39.x86_64 from updates requires authselect >= 1.3, but none of the providers can be installed
  - cannot install both authselect-pr359-325.fc39.x86_64 from copr:copr.fedorainfracloud.org:group_authselect:pr359 and authselect-1.4.3-1.fc39.x86_64 from @System
  - cannot install both authselect-1.4.3-1.fc39.x86_64 from fedora and authselect-pr359-325.fc39.x86_64 from copr:copr.fedorainfracloud.org:group_authselect:pr359
  - conflicting requests
  - cannot install both authselect-pr359-326.fc39.x86_64 from copr:copr.fedorainfracloud.org:group_authselect:pr359 and authselect-1.4.3-1.fc39.x86_64 from @System
  - cannot install both authselect-1.4.3-1.fc39.x86_64 from fedora and authselect-pr359-326.fc39.x86_64 from copr:copr.fedorainfracloud.org:group_authselect:pr359
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages)
~~~

**Comment by pbrezina** [comment]
> Hi @pbrezina do I miss something ?

No, I had to apply https://github.com/authselect/authselect/commit/f023cbc8a108dd9f86d423ca090388a4c668365b so you can upgrade properly, same as we did for sssd.

**Comment by pbrezina** [comment]
Copr images were rebuilt, you can test it.

**Review by justin-stephenson — APPROVED** [review]
Ack, `with-tlog` feature works as expected in testing. Thank you.

**Comment by DaanDeMeyer** [comment]
@pbrezina Note that this change breaks every script that is doing something like `authselect select minimal`.

**Comment by pbrezina** [comment]
True. It is expected that Fedora users will adapt their scripts.

Upgrade to rawhide/F40 should be smooth and automatic - this is handled by rpm scriptlets. New installations gets "local" profile by default, so users needs to change their automation to either remove `authselect select minimal` completely or change it to `authselect select local` or maybe just `authselect enable-feature xyz`. This should be in release notes.

Other distributions may choose to keep packaging the minimal profile if they wish.
